datafusion/physical-expr/src/sort_expr.rs

```diff
@@ -69,4 +69,56 @@ impl PhysicalSortExpr {
             options: Some(self.options),
         })
     }
+
+    pub fn satisfy(&self, requirement: &PhysicalSortRequirements) -> bool {
+        if requirement.sort_options.is_some() {
+            self.options == requirement.sort_options.unwrap()
+                && self.expr.eq(&requirement.expr)
```

Comment:
You can use `If let Some()` idiom here. this would remove `.unwrap()` in the body.